api_types.ts

```diff
@@ -546,6 +547,35 @@ export interface CommonPackFormulaDef<T extends ParamDefs> {
    * @hidden
    */
   readonly allowedAuthenticationNames?: string[];
+
+  /**
+   * The JavaScript function that implements parameter validation.
+   *
+   * This function takes in parameters and a context containing a {@link PermissionSyncMode}
+   * and validates the parameters. A formula may want to validate parameters differently
+   * for permissionSyncMode 'PermissionAware' vs 'Personal' vs undefined (which represents a formula).
+   *
+   * @returns if the parameters are valid
+   * @throws {@link ParameterValidationError} if the parameters are invalid.
+   *
+   * @example
+   * ```
+   * valiateParameters: async function (context, _, formulaContext) {
+   *   let [quantity, sku] = formulaContext?.params;
+   *   let errors = [];
+   *   if (quantity < 0) {
+   *     errors.push({message: "Must be a positive number.", propertyName: "quantity"});
+   *   }
+   *   if (!isValidSku(context, sku)) {
+   *     errors.push({message: `Product SKU not found.`, propertyName: "sku"});
+   *   }
+   *   if (errors.length > 0) {
+   *     throw new ParameterValidationError("Invalid parameter values.", errors);
+   *   }
```

Comment:
This example doesn't return a boolean, do you want to return `true` at the end, or do you prefer if we just make this a void function?